### PR TITLE
Refactor Go Getting Started to use common content

### PR DESCRIPTION
**Pull request by PeloWriter**
This PR refactors the Go content to use the common Getting Started. 

To do:
- review the configuration code sample to ensure it is technically accurate
- add the environment and release variable to the configuration code example, along with code comments
- review the verification example, should it be truncated to simply "sentry.CaptureMessage("It works!")"
- review the content moved to Capture Message and Capture Exception

### CODE REVIEW

File: src/platforms/go/common/goroutines.mdx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Goroutines
 sidebar_order: 5
+description: "Learn more about managing goroutines."
```
Comment (rhcarvalho) [inline src/platforms/go/common/goroutines.mdx:4]: I think we should rename (with a redirect) this section to "Concurrency", as it is about using the SDK when concurrency is at play. 

```suggestion
description: "Learn more about how to use the SDK from concurrent goroutines."
```

File: src/includes/getting-started-primer/go.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<Note>
+<markdown>
+
+_Sentry's Go SDK enables automatic reporting of errors and exceptions._
+
+Our Go SDK supports Go Modules by design. When installed, it picks up the latest version of the SDK and stores it inside the `go.mod` file. Non-major changes should never break your builds. The SDK supports all recent versions of the language, utilizes its features, and gives developers the most helpful hints possible for where and why the error might have occurred.
```
Comment (rhcarvalho) [inline src/includes/getting-started-primer/go.mdx:6]: TODO: will think about a better sentence to put here.

**Review by rhcarvalho — COMMENTED** [review]
I like the new format <3

File: src/includes/getting-started-primer/go.mdx (hunk 2 above)
Comment (rhcarvalho) [reply src/includes/getting-started-primer/go.mdx:6]:
```suggestion
Our Go SDK supports all recent versions of the language, and integrates well with a variety of popular frameworks and packages in the Go ecosystem. It gives developers helpful hints for where and why an error or panic might have occurred.
```

File: src/platforms/go/common/goroutines.mdx (hunk 1 above)
Comment (PeloWriter) [reply src/platforms/go/common/goroutines.mdx:4]: Will do.

File: src/includes/getting-started-install/go.mdx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Install our Go SDK using [`go get`](https://golang.org/cmd/go/#hdr-Module_aware_go_get):
+Install our Go SDK using [`go get`](https://golang.org/cmd/go/#hdr-Module_aware_go_get), and if using Go Modules, it will pick up the latest tagged version:
```
[the remark this answers is not in the document]
Comment (PeloWriter) [reply src/includes/getting-started-install/go.mdx:1]: Updated. Thanks.

**Review by rhcarvalho — APPROVED** [review]

**Pull request merged**